src/Workspaces/Remote/ServiceHub/Services/StackTraceExplorer/RemoteStackTraceExplorerService.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.StackTraceExplorer;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.FindUsages;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+
+namespace Microsoft.CodeAnalysis.Remote.Services.StackTraceExplorer
+{
+    internal sealed class RemoteStackTraceExplorerService : BrokeredServiceBase, IRemoteStackTraceExplorerService
+    {
+        internal sealed class Factory : FactoryBase<IRemoteStackTraceExplorerService>
+        {
+            protected override IRemoteStackTraceExplorerService CreateService(in ServiceConstructionArguments arguments)
+                => new RemoteStackTraceExplorerService(arguments);
+        }
+
+        private int _id;
+
+        public RemoteStackTraceExplorerService(in ServiceConstructionArguments arguments) : base(arguments)
+        {
+        }
+
+        public ValueTask<SerializableDefinitionItem?> TryFindDefinitionAsync(PinnedSolutionInfo solutionInfo, string frameString, StackFrameSymbolPart symbolPart, CancellationToken cancellationToken)
+        {
+            return RunServiceAsync<SerializableDefinitionItem?>(async cancellationToken =>
+            {
+                var solution = await GetSolutionAsync(solutionInfo, cancellationToken).ConfigureAwait(false);
+                if (solution is null)
+                {
+                    throw new InvalidOperationException();
+                }
```

Comment:
should be impossible. do other services do this?

Reply:
No... and I'm surprised there's no warning for a null check that's not needed? This is where quickinfo gets me because it shows `Solution?` for the type